server/lib/analytics/index.js

```diff
@@ -4,15 +4,52 @@
  */
 
 import superagent from 'superagent';
+import { v4 as uuid } from 'uuid';
+import { isUndefined, omit, assign, get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import config from 'config';
 import { statsdTimingUrl } from '../../../client/lib/analytics/statsd';
-import { isUndefined, omit, assign } from 'lodash';
 const URL = require( 'url' );
 
+function getUserFromRequest( request ) {
+	// if user has a cookie, lets use that
+	const encodedUserCookie = get( request, 'cookies.wordpress_logged_in', null );
+
+	if ( encodedUserCookie ) {
+		try {
+			const userCookieParts = decodeURIComponent( encodedUserCookie ).split( '|' );
+
+			// We don't trust it, but for analytics this is enough
+			return {
+				_ul: userCookieParts[ 0 ],
+				_ui: parseInt( userCookieParts[ 1 ], 10 ),
+				_ut: 'wpcom:user_id',
+			};
+		} catch ( ex ) {
+			// ignore the error and let it fallback to anonymous below
+		}
+	}
+
+	// if user doesn't have a cookie, and we had the user identification passed to us on query params
+	// we'll use that, otherwise, we'll just use anonymous.
+	let userIdType = get( request, 'query._ut', 'anon' );
+	let userId = get( request, 'query._ui', null );
+
+	// if user ID not set, even though userIdType is set, we revert to anonymous
+	if ( ! userId ) {
```

Comment:
I think we'd be even safer with something like this:

```
// If we have a full identity, use it
if ( has( request, 'query._ut' ) && has( request, 'query._ui' ) ) {
    return {
        _ui: request.query._ui,
        _ut: request.query._ut
    };
}

// We didn't get a full identity, create an anon ID
return {
    _ut: 'anon',
    _ui: uuid(),
};
```

Though I hope we won't ever send half of the parameters used for a "full" identity, I'd rather we create a new anonymous ID than used `_ui` if `_ut` isn't set: either we have a full proper identity in `_ui` and `_ut`, or we discard it and create a proper "anonymous" identity?

Reply:
Yep, you're right again, I've updated the code.